elementary/clients/gcs/client.py

```diff
@@ -19,15 +19,23 @@ def __init__(self, config: Config):
     def create_client(cls, config: Config) -> Optional['GCSClient']:
         return cls(config) if config.has_gcs else None
 
-    def send_report(self, html_path: str) -> bool:
+    # html_path is the local path where the report generated at.
+    # bucket_file_path is where we want to upload the report to at the bucket - support folders!
+    def send_report(self, html_path: str, bucket_file_path: Optional[str] = None) -> bool:
```

Comment:
I'd rename html_path to local_html_file_path and bucket_file_path to remote_bucket_file_path so it will be really clear in the code itself (then you can remove the comment above)